src/docfx/build/xref/ExternalXrefMapLoader.cs

```diff
@@ -30,8 +30,7 @@ internal class ExternalXrefMapLoader
                 }
                 else
                 {
-                    // TODO: It is very easy to forget passing fallbackDocsetPath, the RestoreMap interface needs improvement
-                    var filePath = RestoreMap.GetRestoredFilePath(docset.DocsetPath, url, docset.FallbackDocset?.DocsetPath);
+                    var filePath = RestoreFileMap.GetRestoredFilePath(docset, url);
```

Comment:
It is possible to abstract away all the unrelated concept, so the caller only needs to pass in an `url`?

```suggestion
                    var filePath = xxx.GetFileRestorePath(url);
```